django/contrib/sitemaps/views.py

```diff
@@ -65,16 +66,23 @@ def sitemap(request, sitemaps, section=None,
                 site = site()
             urls.extend(site.get_urls(page=page, site=req_site,
                                       protocol=req_protocol))
+            if hasattr(site, 'latest_lastmod'):
+                site_lastmod = site.latest_lastmod
+                if type(site_lastmod) is datetime.date:
+                    site_lastmod = datetime.datetime(site_lastmod.year,
+                                                     site_lastmod.month,
+                                                     site_lastmod.day)
+                if lastmod is None or lastmod < site_lastmod:
+                    lastmod = site_lastmod
         except EmptyPage:
             raise Http404("Page %s empty" % page)
         except PageNotAnInteger:
             raise Http404("No page '%s'" % page)
     response = TemplateResponse(request, template_name, {'urlset': urls},
                                 content_type=content_type)
-    if hasattr(site, 'latest_lastmod'):
-        # if latest_lastmod is defined for site, set header so as
+    if lastmod is not None:
+        # if lastmod is defined for at least one site, set header so as
```

Comment:
I have to agree with you here. The code is rewritten to explicitly require having lastmod defined for all sitemaps.